jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java

```diff
@@ -104,6 +97,77 @@ private RoutePort createRoutePort(ServiceBuilder serviceBuilder) {
         return routePort;
     }
 
+    private String prepareHostForRoute(String routeDomainPostfix, String name) {
+        String ret = FileUtil.stripPostfix(name,"-service");
+        ret = FileUtil.stripPostfix(ret,".");
+        ret += ".";
+        ret += FileUtil.stripPrefix(routeDomainPostfix, ".");
+        return ret;
+    }
+
+    private Set<Integer> getPorts(ServiceBuilder service) {
+        Set<Integer> answer = new HashSet<>();
+        if (service != null) {
+            ServiceSpec spec = getOrCreateSpec(service);
+            for (ServicePort port : spec.getPorts()) {
+                answer.add(port.getPort());
+            }
+        }
+        return answer;
+    }
+
+    public static ServiceSpec getOrCreateSpec(ServiceBuilder entity) {
+        ServiceSpec spec = entity.buildSpec();
+        if (spec == null) {
+            spec = new ServiceSpec();
+            entity.editOrNewSpec().endSpec();
+        }
+        return spec;
+    }
+
+    private boolean hasExactlyOneServicePort(ServiceBuilder service, String id) {
+        Set<Integer> ports = getPorts(service);
+        if (ports.size() != 1) {
+            log.info("Not generating route for service " + id + " as only single port services are supported. Has ports: " +
+                    ports);
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    private void addRoute(KubernetesListBuilder listBuilder, ServiceBuilder serviceBuilder, List<Route> routes) {
+        ObjectMeta serviceMetadata = serviceBuilder.buildMetadata();
+
+        if (serviceMetadata != null && StringUtils.isNotBlank(serviceMetadata.getName())
+                && hasExactlyOneServicePort(serviceBuilder, serviceMetadata.getName()) && isExposedService(serviceMetadata, log)) {
+            String name = serviceMetadata.getName();
+            if (!hasRoute(listBuilder, name)) {
+                if (StringUtils.isNotBlank(routeDomainPostfix)) {
+                    routeDomainPostfix = prepareHostForRoute(routeDomainPostfix, name);
+                } else {
+                    routeDomainPostfix = "";
+                }
+
+                RoutePort routePort = createRoutePort(serviceBuilder);
+                if (routePort != null) {
+                    // TODO one day lets support multiple ports on a Route when the model supports it
+                    RouteBuilder routeBuilder = new RouteBuilder().
+                            withMetadata(serviceMetadata).
+                            withNewSpec().
+                            withPort(routePort).
+                            withNewTo().withKind("Service").withName(name).endTo().
+                            withHost(routeDomainPostfix.isEmpty() ? null : routeDomainPostfix).
+                            endSpec();
+
+                    // removing `expose : true` label from metadata.
```

Comment:
What does this imply?

Reply:
I think it's about removing `expose=true` labels added by [ExposeEnricher](https://github.com/eclipse/jkube/blob/master/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/ExposeEnricher.java)

Reply:
Yes, I see that. But what does that expose=true label > ExposeEnricher do?

I think it something fabric8 only related (hence deprecated) but I may be wrong

https://blog.fabric8.io/additional-kubernetes-controllers-from-fabric8-you-can-use-with-your-microservice-3126a2c4c132
https://github.com/jenkins-x/exposecontroller